lib/src/commands/music.dart

```diff
@@ -0,0 +1,161 @@
+import 'package:running_on_dart/src/checks.dart';
+import 'package:running_on_dart/src/services/music.dart';
+
+ChatGroup music = ChatGroup(
+  'music',
+  'Music related commands',
+  checks: [
+    GuildCheck.all(),
+    sameVoiceChannelOrDisconnectedCheck,
+  ],
+  children: [
+    ChatCommand(
+      'play',
+      'Plays music based on the given query',
+      id('music-play', (
+          IChatContext context,
+          @Description('The name/url of the song/playlist to play') String query
+        ) async {
+        final node = MusicService.instance.cluster.getOrCreatePlayerNode(context.guild!.id);
+        connectIfNeeded(context);
+        final result = await node.autoSearch(query);
+
+        if (result.tracks.isEmpty) {
+          await respond(context, MessageBuilder.content('No results were found'));
+        }
+
+        if (result.playlistInfo.name != null) {
+          for (final track in result.tracks) {
+            node.play(
+              context.guild!.id,
+              track,
+              requester: context.member!.id,
+              channelId: context.channel.id
+            ).queue();
+          }
+
+          await respond(context, MessageBuilder.content(
+              'Playlist `${result.playlistInfo.name}`($query) enqueued'
+          ));
+
+        } else {
+          node.play(
+            context.guild!.id,
+            result.tracks[0],
+            requester: context.member!.id,
+            channelId: context.channel.id
+          ).queue();
+          await respond(context, MessageBuilder.content(
+              'Track `${result.tracks[0].info?.title}` enqueued'
+          ));
+        }
+      })
+    ),
+    ChatCommand(
+      'skip',
+      'Skips the currently playing track',
+      checks: [connectedToAVoiceChannelCheck],
+      id('music-skip', (IChatContext context) async {
+        final node = MusicService.instance.cluster.getOrCreatePlayerNode(context.guild!.id);
+        final player = node.players[context.guild!.id]!;
+
+        if (player.queue.isEmpty) {
+          await respond(context, MessageBuilder.content('The queue is clear!'));
+          return;
+        }
+
+        node.skip(context.guild!.id);
+        await respond(context, MessageBuilder.content('Skipped current track'));
+      })
+    ),
+    ChatCommand(
+      'stop',
+      'Stops the current player and clears its track queue',
+      checks: [connectedToAVoiceChannelCheck],
+      id('music-stop', (IChatContext context) async {
+        final node = MusicService.instance.cluster.getOrCreatePlayerNode(context.guild!.id);
+        node.stop(context.guild!.id);
+        await respond(context, MessageBuilder.content('Player stopped!'));
+      })
+    ),
+    ChatCommand(
+      'leave',
+      'Leaves the current voice channel',
+      checks: [connectedToAVoiceChannelCheck],
+      id('music-leave', (IChatContext context) async {
+        (context.client as INyxxWebsocket)
```

Comment:
Why not just use [`context.guild!.shard`](https://pub.dev/documentation/nyxx/latest/nyxx/IGuild/shard.html)?